pkg/karmadactl/cmdinit/kubernetes/deploy.go

```diff
@@ -180,8 +181,16 @@ func (i *CommandInitOption) validateLocalEtcd(parentCommand string) error {
 		return fmt.Errorf("when etcd storage mode is hostPath, dataPath is not empty. See '%s init --help'", parentCommand)
 	}
 
-	if i.EtcdStorageMode == etcdStorageModeHostPath && i.EtcdNodeSelectorLabels != "" && utils.StringToMap(i.EtcdNodeSelectorLabels) == nil {
-		return fmt.Errorf("the label does not seem to be 'key=value'")
+	if i.EtcdStorageMode == etcdStorageModeHostPath && i.EtcdNodeSelectorLabels != "" {
+		selector, err := metav1.ParseToLabelSelector(i.EtcdNodeSelectorLabels)
+		if err != nil {
+			return fmt.Errorf("the etcdNodeSelector format is incorrect: %s", err)
+		}
+		labelMap, err := metav1.LabelSelectorAsMap(selector)
+		if err != nil {
+			return fmt.Errorf("failed to convert etcdNodeSelector labels to map: %v", err)
+		}
+		i.EtcdNodeSelectorLabelsMap = labelMap
```

Comment:
I have a question, and the variable setting logic is involved here, is it more reasonable to put it in the Complete function?

Reply:
I think this is make sense,I'm glad we're getting closer and closer to merge this PR.